app/src/componentsV2/CodeEditor/components/EditorV2/Editor.tsx

```
@@ -0,0 +1,361 @@
+          dispatch(globalActions.updateIsCodeEditorFullScreenModeOnboardingCompleted(true));
+        }
+      }
+    } else {
+      trackCodeEditorCollapsedClick(analyticEventProperties);
+    }
+  };
+
+  const editorLanguage = useMemo(() => {
+    switch (language) {
+      case EditorLanguage.JAVASCRIPT:
+        return javascript({ jsx: false });
+      case EditorLanguage.JSON:
+        return json();
+      case EditorLanguage.HTML:
+        return html();
+      case EditorLanguage.CSS:
+        return css();
+      default:
+        return null;
+    }
+  }, [language]);
+
+  const editorRefCallback = (editor: ReactCodeMirrorRef) => {
+    if (!editorRef.current && editor?.editor && editor?.state && editor?.view) {
+      editorRef.current = editor;
+      setIsEditorInitialized(true);
+    }
+  };
+
+  const updateContent = useCallback((prettifiedCode: string) => {
+    if (editorRef.current?.view) {
+      const view = editorRef.current.view;
+      const transaction = view.state.update({
+        changes: { from: 0, to: view.state.doc.length, insert: prettifiedCode },
+      });
+      view.dispatch(transaction);
+    }
+  }, []);
+
+  const applyPrettification = useCallback(() => {
+    if (config?.enablePrettify) {
+      if (language === EditorLanguage.JSON || language === EditorLanguage.JAVASCRIPT) {
+        const prettified = prettifyCode(value, language);
+        updateContent(prettified.code);
+      }
+    }
+  }, [config?.enablePrettify, language, value, updateContent]);
+
+  useEffect(() => {
+    if (isEditorInitialized && !isDefaultPrettificationDone.current) {
+      applyPrettification();
+      isDefaultPrettificationDone.current = true;
+    } else if (!prettifyOnInit) {
+      //on render
+      applyPrettification();
+    }
+  }, [isEditorInitialized, isDefaultPrettificationDone, applyPrettification, prettifyOnInit]);
+
+  const handleEditorClose = useCallback(
+    (id: string) => {
+      // @ts-expect-error
```

Can you please add some context here? Why expect error?

---

```
// @ts-expect-error
dispatch(globalActions.removeToastForEditor({ id }));
```

- `removeToastForEditor` reducer function is in `editorToastAction.js` , so to suppress that error from being reported in `Editor.tsx` file 
- To remove this need to changes editorToastAction to `editorToastAction.ts`